basic edits to section 1 from policy authority doc based edits

lachellel opened the pull request:
#9 
@weirdscience @techliaison @LarryFrank @konklone 

- Added 75% of initial edits from draft provided to @ChiHickey
- In a few scenarios, used ~~strike through~~ to maintain original and show _in-line_ edits 
  - can grep and remove later after editing 

- For title of "Federal PKI" - defer to contributors.  I added Federal PKI per the edits.  No preference and can change.

- All definitions and acronyms have not changed

- Maintained Notice to Readers; included updates; and maintained licensing and contributing

lachellel committed:
basic edits to section 1 from policy authority doc based edits

lachellel committed:
trying to fix merge conflicts 

from demo pull requests

lachellel commented:
@weirdscience @techliaison @LarryFrank @konklone @twbaldridge

this open pull request is for section 1 and basic updates.  All items can be modified with additional conversations and pull requests

I'd like to merge these changes - agree or disagree?

LarryFrank commented on certificate-policy.md:
```diff
@@ -1,46 +1,43 @@
 
 ## 1.1 Overview
-This CP describes an integrated set of technologies, protocols, identity-proofing, lifecycle management, and auditing requirements that are necessary (but not sufficient) for the issuance and management of Publicly-Trusted Certificates; Certificates that are trusted by virtue of the fact that their corresponding Root Certificate is distributed in widely-available application software. The require ...
+This CP describes an integrated set of technologies, protocols, identity-proofing, lifecycle management, and auditing requirements that are necessary (but not sufficient) for the issuance and management of US Federal issued Publicly-Trusted Certificates; Certificates that are trusted by virtue of the fact that their corresponding Root Certificate is distributed in widely-available application sof ...
```
What is the intent of "(but not sufficient)" and the "subset of requirements" in line 27 below.  Will there be some additional document that fills in the rest?  My preference - the CP should be as complete as possible in order to ensure consistent implementation.

techliaison replied:
I agree with Larry - in addition,  I would propose dropping this sentence:
"The requirements are not mandatory for Certification Authorities unless and until they become adopted and enforced by relying-party Application Software Suppliers"
we should not be putting any requirements into our CP that we do not want to make mandatory on the CAs operating under this CP - the number of CAs we expect to be operating under this new CP is small and we should not be dependent on enforcement by the RPs for mandating issuance rules.

LarryFrank commented on certificate-policy.md:
```diff
@@ -1,46 +1,43 @@
+The CP for the Issuance and Management of US Federal issued Publicly-Trusted Certificates describe a subset of the requirements that a Certification Authority must meet in order to issue Publicly Trusted Certificates. This document serves two purposes:  to specify Federal PKI requirements and to provide guidance and requirements for what a CA must address in its CPS.  Except where explicitly stat ...
 
-These Requirements only address Certificates intended to be used for authenticating servers accessible through the Internet. Similar requirements for code signing, S/MIME, time-stamping, VoIP, IM, Web services, etc. may be covered in future versions.
+These Requirements do not address all of the issues relevant to the issuance and management of US Federal issued Publicly-Trusted Certificates. In accordance with RFC 3647 and to facilitate a comparison of other certificate policies and CPSs (e.g. for policy mapping), this CP includes all sections of the RFC 3647 framework.  The Federal PKI Policy Authority may update these Requirements from time ...
```
> ~~ In particular, it is expected that a future version will contain more formal and comprehensive audit requirements for delegated functions.~~ 

As above, I think the CP needs to address the formal and comprehensive audit requirements.

LarryFrank commented on certificate-policy.md:
```diff
@@ -171,16 +117,22 @@ This Certificate Policy for Baseline Requirements for the Issuance and Managemen
 
 ### 1.5.4 CPS approval procedures
-No stipulation.
+Independent Auditors conduct assessments of CPS conformance to the CP requirements.  The results of these audits are submitted and approved by the Federal Public Key Infrastructure Policy Authority.
```
I know this is not the way the Fed PKI works but having the compliance auditor do compliance analysis is too late.  Need an approved CPS to stand up operation - audit happens (at best) just prior to commencing operation - at which time, it may be very expensive to fix thing.  DoD has someone actually do a compliance analysis before the PMA approves the CPS - then the auditor checks against that.  Frankly, from what I have seen looking at several fed agency CPS, compliance auditors miss a lot...

lachellel replied:
issue #24 opened to discuss

techliaison commented on certificate-policy.md:
```diff
@@ -1,46 +1,43 @@
+The CP for the Issuance and Management of US Federal issued Publicly-Trusted Certificates describe a subset of the requirements that a Certification Authority must meet in order to issue Publicly Trusted Certificates. This document serves two purposes:  to specify Federal PKI requirements and to provide guidance and requirements for what a CA must address in its CPS.  Except where explicitly stat ...
 
-These Requirements only address Certificates intended to be used for authenticating servers accessible through the Internet. Similar requirements for code signing, S/MIME, time-stamping, VoIP, IM, Web services, etc. may be covered in future versions.
+These Requirements do not address all of the issues relevant to the issuance and management of US Federal issued Publicly-Trusted Certificates. In accordance with RFC 3647 and to facilitate a comparison of other certificate policies and CPSs (e.g. for policy mapping), this CP includes all sections of the RFC 3647 framework.  The Federal PKI Policy Authority may update these Requirements from time ...
```
Why would we leave out issues relevant to the issuance & management of the US Federal issued Pubicly-Trusted Certificates?
I believe the old text more correctly stated the intent of this CP.
Also leave out the In particular, it is expected that a future version will contain more formal and comprehensive audit requirements for delegated functions.
If we expect there to be delegated functions like validation services we need to defie the requriements in order to have consistent issuing practices.

lachellel replied:
I was just the sherpa!  I copied directly from the word version marked up and provided to me!   Reverting back to original and resubmitting...

lachellel replied:
reverted back / PR updated

lachellel committed:
removing not sufficient

lachellel committed:
remove "in particular"

techliaison commented on certificate-policy.md:
```diff
@@ -54,86 +51,35 @@ The following Certificate Policy identifiers are reserved for use by CAs as an o
 2. Retain documentation in accordance with Section 5.5.2;
 3. Abide by the other provisions of these Requirements that are applicable to the delegated function; and
 4. Comply with (a) the CA's Certificate Policy/Certification Practice Statement or (b) the Delegated Third Party's practice statement that the CA has verified complies with these Requirements.
+5. Agree on which party is responsible for the annual audit of the delegated functions.
```
we need to "federalize" this example since we will be constrained to .gov/.mil - examples of companies are out of scope:
certificate containing the Subject name "XYZ Co." on the authority of Enterprise RA "ABC Co.", unless the two companies are affiliated (see Section 3.2) or "ABC Co." is the agent of "XYZ Co". This requirement applies regardless of whether the accompanying requested Subject FQDN falls within the Domain Namespace of ABC Co.'s Registered Domain Name.

maybe 
certificate containing the nae of "Agency1" on the authority of an "Agency2" RA, unless the two agencies have a system sharing agreement. This requirement applies regardless of whether the accompanying requested Subject FQDN falls within the Domain Namespace of "Agency2's" Registered Domain Name.

LarryFrank replied:
Agree we should federalize - but DoD needs some specific .EDU (National Defense Universities, military academies) - can constrain them to domains held by DoD...

lachellel replied:
Updated example in PR

domain constraints in later sections

>DoD needs some specific .EDU (National Defense Universities, military academies)

techliaison commented on certificate-policy.md:
```diff
@@ -150,7 +96,7 @@ As defined in Section 1.6.1.
+"Relying Party" and "Application Software Supplier" are defined in Section 1.6.1. 
 
 
 ### 1.3.5 Other Participants
```
shouldn't we mention independent auditors as one type of "other participant"?

lachellel replied:
Currently, the other participants text is specific to the BRD development.  I think we should update AFTER we iterate, to ensure inclusiveness.  

> ### 1.3.5 Other Participants
> Other groups that have participated in the development of these Requirements include the AICPA/CICA WebTrust for Certification Authorities task force and ETSI ESI. Participation by such groups does not imply their endorsement, recommendation, or approval of the final product.

LarryFrank commented on certificate-policy.md:
```diff
@@ -54,86 +51,35 @@ The following Certificate Policy identifiers are reserved for use by CAs as an o
 
 ### 1.3.2 Registration Authorities
-The CA MAY delegate the performance of all, or any part, of Section 3.2 requirements to a Delegated Third Party, provided that the process as a whole fulfills all of the requirements of Section 3.2.
+The CA MAY delegate the performance of all, or any part, of Section 3.2 requirements to a Delegated Third Party, provided that the process as a whole fulfills all of the requirements of Sections 3.2 and 8.
```
In DoD (and the rest of the FED, I think) the typical "delegated third parties" are called "Registration Authorities" - and typically are part of the organization (certainly Federal Government or government contractors) and possibly Certificate Status Authorities.   Rather than fuzzy language that needs further interpretation - can we use specific language that says what we will do?  Goes to my question about whether we really need to publish a CPS is the CP has the right level of detail.

techliaison replied:
agree

lachellel replied:
Intent to publish both a CP and a CPS "template"

For replacing the Delegated Third Party with Registration Authority - wouldn't this create MORE confusion?   Since an RA is referenced as an **Enterprise RA** with separate responsibilities?

> The CA MAY designate an Enterprise RA to verify certificate requests from the Enterprise RA's own organization.

we definitely use "RA" in many different contexts in Fed (RA as a service boundary, RA as a person / program, LRA, etc)

techliaison commented on certificate-policy.md:
```diff
@@ -1,46 +1,43 @@
-
+This certificate policy (CP) contains the requirements for the issuance and management of publicly-trusted SSL certificates, as adopted by the CA/Browser Forum and tailored for use within the US Federal Government.
 
 The following Certificate Policy identifiers are reserved for use by CAs as an optional means of asserting compliance with this CP (OID arc 2.23.140.1.2) as follows:
```
Do we want to also include an FPKI policy OID to indicate federally issued?
We just added a cert policy for publicly-trusted-serverAuth certs to the Common Policy CP, I think it would be better to define it in this CP since we decided to create a separate CP for publicly trusted federally issued TLS certificates.

LarryFrank replied:
Adding an OID that says this was issued by a Federal CA is redundant - the Federal CA that issued it is identified in the certificate.  Any relying party that cares can determine that directly by looking at the path to the trusted root.

I was and remain confused about why we added the "publicly-trusted-serverAuth certs " to Common...

lachellel replied:
opened issue #27 to discuss

lachellel committed:
updating Section 1.2 example for "agencies"

Commercial example updated to be federal example

lachellel commented:
I either opened an Issue or updated the PR to address the comments!

Would like a merge and we can review / iterate again in another PR?

lachellel committed:
Prohibited uses / policy administration

Draft updates for Section 1.4.1 and 1.4.2
Suggested per #26

Included draft update for Section 1.5 to include the USG
